src/engine/TransitivePath.cpp

```diff
@@ -582,35 +392,20 @@ void TransitivePath::computeTransitivePathRightBound(
   // be modified after this point.
   std::vector<std::shared_ptr<const ad_utility::HashSet<Id>>> edgeCache;
 
-  Id last_elem = ID_NO_VALUE;
-  size_t last_result_begin = 0;
-  size_t last_result_end = 0;
-  for (size_t i = 0; i < right.size(); i++) {
-    checkTimeoutHashSet();
-    if (right[i][rightSideCol] == last_elem) {
-      // We can repeat the last output
-      size_t num_new = last_result_end - last_result_begin;
-      size_t res_row = res.size();
-      res.resize(res.size() + num_new);
-      for (size_t j = 0; j < num_new; j++) {
-        for (size_t c = 0; c < resWidth; c++) {
-          res(res_row + j, c) = res(last_result_begin + j, c);
-        }
-      }
-      checkTimeoutAfterNCalls(num_new * resWidth);
+  for (size_t i = 0; i < nodes.size(); i++) {
```

Comment:
I think this can be `for (Id currenStartNode : startNodes)`. 
The only other usage is the call to `marks.clear` which can be moved to the beginning of the node.